demos/bubble_shear.py

```diff
@@ -0,0 +1,237 @@
+        form_fields = {"c": (c, c_), "u": (u, u_)}
+        F = mesh_seq.form(index, form_fields)["c"]
+        nlvp = NonlinearVariationalProblem(F, c, bcs=mesh_seq.bcs(index))
+        nlvs = NonlinearVariationalSolver(nlvp, ad_block_tag="c")
+
+        # Time integrate from t_start to t_end
+        t = t_start + dt
+        while t < t_end + 0.5 * dt:
+            # update the background velocity field at the current timestep
+            u.interpolate(velocity_expression(x, y, t))
+
+            # solve the advection equation
+            nlvs.solve()
+
+            # update the 'lagged' concentration and velocity field
+            c_.assign(c)
+            u_.assign(u)
+            t += dt
+
+        return {"c": c}
+
+    return solver
+
+
+# Now that we have defined the solver and specified how to build and update the
+# background velocity, we are ready to solve the problem. We run the simulation
+# until :math:`t=T/2` on a sequence of two coarse meshes. ::
+
+# Reduce the cost of the demo during testing.
+test = os.environ.get("GOALIE_REGRESSION_TEST") is not None
+n = 50 if not test else 10
+dt = 0.01 if not test else 0.025
+
+num_subintervals = 2
+meshes = [UnitSquareMesh(n, n) for _ in range(num_subintervals)]
+end_time = period / 2
+time_partition = TimePartition(
+    end_time,
+    len(meshes),
+    dt,
+    fields,
+    num_timesteps_per_export=30,
+)
+
+msq = MeshSeq(
+    time_partition,
+    meshes,
+    get_function_spaces=get_function_spaces,
+    get_initial_condition=get_initial_condition,
+    get_bcs=get_bcs,
+    get_form=get_form,
+    get_solver=get_solver,
+)
+solutions = msq.solve_forward()
+
+# Let us plot the solution :math:`c` at exported timesteps. ::
+
+plot_kwargs = {"cmap": "coolwarm", "levels": 100}
+fig, _, _ = plot_snapshots(solutions, time_partition, "c", "forward", **plot_kwargs)
+fig.savefig("bubble_shear.jpg")
+
+# .. figure:: bubble_shear.jpg
+#    :figwidth: 80%
+#    :align: center
+#
+# We observe that the initial tracer bubble has been deformed by the strong shear forces.
+# However, we notice that the deformation appears to be reverting. This is due to the
+# periodicity of the velocity field :math:`\mathbf{u}`, which we chose to be an odd
+# function of time. This means that the velocity field reverses direction at :math:`t=T/2`,
+# and so does the deformation of the tracer bubble, which returns to its initial shape.
+#
+# However, while the tracer bubble at :math:`t=T/2` does resemble a very blurry circular
+# shape, it is far from matching the initial condition. This "bluriness" is the result of
+# adding the SUPG stabilisation to the weak form, which adds numerical diffusion. This
+# diffusion is necessary for numerical stability and for preventing oscillations, but it
+# also makes the solution irreversible. The amount of difussion added is related to the
+# grid Peclet number :math:`Pe = U\,h/2D`: the coarser the mesh is, the more diffusion
```

Comment:
Minor point: "Péclet number :math:`\mathrm{Pe} = U\,h/2D`"

Reply:
Fixed in 262c78a51a99714a4794b55259edf6fc5aea3567.